.github/actions/has-auto-releasable-commits-v1/src/shouldAutoRelease.test.ts

```diff
@@ -0,0 +1,227 @@
+import 'mocha'
+import { assert } from 'chai'
+import shouldAutoRelease from './shouldAutoRelease'
+
+describe('shouldAutoRelease', () => {
```

Comment:
This testing suite might be able to be simplified / more readable by using a test suite matrix with the different input types and expected output. Something like:

```js
const testMatrix = [
  {
    // not versioned locked with patch change
    versionLocked: false,
    commitType: 'fix',
    expected: true
  },
  {
    // version locked with patch change and axe-core patch
    versionLocked: true,
    axeCoreType: 'fix',
    commitType: 'fix',
    expected: true
  },
  {
    // no commits
    versionLocked: true,
    expected: false
  }
]
```

Reply:
I tried doing this, it becomes quite annoying when we start nesting cases, instead I've removed the unnecessary commit objects into: 

`createCommit` and `createAxeCoreCommit`

```ts
describe('and version is locked', () => {
      describe('and there are no breaking changes', () => {
        describe('and there are no major or minor changes for axe-core', () => {
          describe('and there is a feat commit', () => {
            it('should return true', () => {
              const result = shouldAutoRelease({
                commitList: [createCommit('feat')],
                isVersionLocked: true
              })

              assert.isTrue(result)
            })
          })

          describe('and there is a fix commit', () => {
            it('should return true', () => {
              const result = shouldAutoRelease({
                commitList: [createCommit('fix')],
                isVersionLocked: true
              })

              assert.isTrue(result)
            })
          })
        })
      })
})
```

Now should be easier to read the test case as:

```
When there are commits, and the version is locked, and there are no breaking changes, and there are no major or minor releases for axe-core, and there is a feat commit, returns true

// and so on
```